cmd/kubeadm/app/cmd/init.go

```diff
@@ -155,6 +156,10 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 		"Port for JWS discovery service to bind to",
 	)
 
+	cmd.PersistentFlags().StringVar(
+		&cfg.ImagePrefix, "image-prefix", gcrPrefix,
+		"use specific images instead of gcr",
```

Comment:
This help string should be more descriptive.  I'd do something like: `Specify an alternate location for all images.  Currently this includes: etcd, kube-discovery, kube-apiserver, kube-controller-manager, kube-scheduler, kube-proxy, kubedns, kube-dnsmasq, exechealthz, pause`

Ideally that list would be generated and we'd have clear instructions on how to create a custom set of images.  Or we'd have a way to override just some of them as necessary.  As things stand, this is a pretty advanced move.  Unless we are ready to document how to use this we should probably make it an env variable instead of a command line flag in order to not pollute our `kubeadm --help`.